lib/web_ui/test/engine/history_test.dart

```diff
@@ -0,0 +1,263 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```

Comment:
@nturgut FYI, this test is good first candidate for a cross framework-engine integration test. Currently it has to emulate what the framework does, which could go out of sync with the framework. But an integration test would allow us to actually have the framework respond to and initiate history changes. That would make it a much more reliable test.